src/coordinator/functions/clamp.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package functions
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/m3db/m3db/src/coordinator/block"
+	"github.com/m3db/m3db/src/coordinator/executor/transform"
+	"github.com/m3db/m3db/src/coordinator/parser"
+)
+
+// ClampMinType ensures all values except NaNs are greater than or equal to the provided argument
+const ClampMinType = "clamp_min"
+
+// ClampMaxType ensures all values except NaNs are lesser than or equal to provided argument
+const ClampMaxType = "clamp_max"
+
+// ClampOp stores required properties for clamp
+type ClampOp struct {
+	optype string
+	scalar float64
+}
+
+// NewClampOp creates a new clamp op based on the type and arguments
+func NewClampOp(args []interface{}, optype string) (ClampOp, error) {
+	if len(args) != 1 {
```

Comment:
If we require len to be 1, why not just have `interface{}` instead of `[]interface{}` in the function definition?

Reply:
so if you look at the parser, the idea is that parser doesn't know exactly how to provide the args, each function figures out how it uses its args